bn_s_mp_check_endianness.c

```diff
@@ -0,0 +1,25 @@
+#include "tommath_private.h"
+#ifdef BN_S_MP_CHECK_ENDIANNESS_C
+/* LibTomMath, multiple-precision integer library -- Tom St Denis */
+/* SPDX-License-Identifier: Unlicense */
+
+mp_endian s_mp_check_endianness(void)
+{
+   mp_endian endian;
+   int16_t n;
+   char *p;
+   /*
+       This sorts out many, if not all of the "odd" endiannesses
+       but also modern DSPs with CHAR_BIT > 8.
+    */
+   if (CHAR_BIT != 8) {
+      return MP_UNKNOWN_ENDIAN;
```

Comment:
No. Rather add a static_assert for charbit==8

Reply:
Yes, I forgot that we even have a macro for that.